python/src/iceberg/table/partitioning.py

```diff
@@ -64,3 +67,100 @@ def __str__(self):
 
     def __repr__(self):
         return f"PartitionField(field_id={self.field_id}, name={self.name}, transform={repr(self.transform)}, source_id={self.source_id})"
+
+    def __hash__(self):
+        return hash((self.source_id, self.field_id, self.name, self.transform))
```

Comment:
Why is `__hash__` needed? Are partition fields used as map keys?

Reply:
yeah I added back because it's being used in the hash of the PartitionSpec at line 129

Reply:
> because it's being used in the hash of the PartitionSpec at line 129

Why does `PartitionSpec` define `__hash__`? Will spec be used as a key in a dict or map? I doubt it so I'd probably remove both.